src/pytorch_lightning/loops/progress.py

```diff
@@ -259,31 +259,22 @@ class OptimizationProgress(BaseProgress):
 
     Args:
         optimizer: Tracks optimizer progress.
-        optimizer_position: The index of the current optimizer amongst the currently active optimizers.
```

Comment:
TODO: migration in checkpoint

Reply:
why? having leftover keys should be fine, no? And since 2.0 is breaking , we don't need 2.0 checkpoints to be supported with 1.9.X

Reply:
I think it's better to clean things up. 
In a follow up, we will probably collapse the optimizer loop to flatten the hierarchy. In this case, we would extend this migration here to move the optimizer progress tracking state up one level (similar to tbptt). A remapping of the key would anyway be necessary, and then this clean up here would be just an extra.